prymer/primer3/primer3_parameters.py

```diff
@@ -62,19 +62,61 @@ class stores user input for internal probe design and maps it to the correct Pri
 
 ```
 """
-
-import warnings
+import typing
+from abc import ABC, abstractmethod
 from dataclasses import dataclass
 from dataclasses import fields
 from typing import Any
 from typing import Optional
 
-from prymer.model import MinOptMax
+from prymer.model import MinOptMax, Span
 from prymer.primer3.primer3_input_tag import Primer3InputTag
+from prymer.primer3.primer3_task import Primer3TaskType
+
+
+class Primer3Parameters(ABC):
+    target: Span
+    task: Primer3TaskType
+
+    def as_amplicon_params(self) -> "AmpliconParameters":
+        """Use this method when you want to treat these parameters as amplicon parameters."""
+        if isinstance(self, AmpliconParameters):
+            return typing.cast(AmpliconParameters, self)
+        raise Exception("The parameters are not amplicon parameters")
+
+    def as_probe_params(self) -> "ProbeParameters":
+        """Use this method when you want to treat these parameters as probe parameters."""
+        if isinstance(self, ProbeParameters):
+            return typing.cast(ProbeParameters, self)
+        raise Exception("The parameters are not amplicon parameters")
```

Comment:
This feels out of place to me - why not just check and cast where that's needed rather than having it in the superclass?